### PR TITLE
Refactor how HOPs create new args to subgraphs

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115206
* #115187
* #115186
* __->__ #115185

This PR combines the logic for Tensor and SymNode.

Test Plan:
- Existing tests

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/higher_order_ops.py [hunk 1]
```diff
@@ -131,30 +131,29 @@ def validate_args_and_maybe_create_graph_inputs(
...
+                tracer.create_graph_input(a.as_proxy().node.name)
+            new_arg = a
+        # If `a` can be put into a graph
+        elif a.maybe_fx_node() is not None:
             if manually_set_subgraph_inputs:
-                new_proxy = tracer.create_graph_input(str(a.sym_num.node.expr))
+                node = a.maybe_fx_node()
+                new_proxy = tracer.create_graph_input(node.name)
+                example_value = (
+                    node.meta["example_value"] if "example_value" in node.meta else None
```
Comment (ydwu4) [inline torch/_dynamo/variables/higher_order_ops.py:146]: When "example_value" is None? Is it the SymBool inputs logic in builder.py, which doesn't set the example value

**Review by ydwu4 — APPROVED** [review]
Looks good!

File: torch/_dynamo/variables/higher_order_ops.py (hunk 1 above)
Comment (zou3519) [reply torch/_dynamo/variables/higher_order_ops.py:146]: I'm not sure, I just put this here just in case it is possible

**Review by yanboliang — APPROVED** [review]

File: torch/_dynamo/variables/higher_order_ops.py (hunk 1 above)
Comment (yanboliang) [reply torch/_dynamo/variables/higher_order_ops.py:146]: Dynamo should produce ```example_value``` during fake tensor propagation if everything works well 😄.

**Pull request closed without merging**